Marketplace/voiceHUDApp/App Resources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,815 @@
+    var AVERAGING_RATIO = 0.05,
+        LOUDNESS_FLOOR = 11.0,
+        LOUDNESS_SCALE = 2.8 / 5.0,
+        LOG2 = Math.log(2.0),
+        AUDIO_PEAK_DECAY = 0.02;
+
+    var audio = {
+        update: function (uuid) {
+            var user = userStore[uuid];
+            if (!user) {
+                console.log("not a user");
+                return;
+            }
+
+            // scale audio
+            function scaleAudio(val) {
+                var audioLevel = 0.0;
+                if (val <= LOUDNESS_FLOOR) {
+                    audioLevel = val / LOUDNESS_FLOOR * LOUDNESS_SCALE;
+                } else {
+                    audioLevel = (val - (LOUDNESS_FLOOR - 1)) * LOUDNESS_SCALE;
+                }
+                if (audioLevel > 1.0) {
+                    audioLevel = 1;
+                }
+                return audioLevel;
+            }
+
+            // the VU meter should work similarly to the one in AvatarInputs: log scale, exponentially averaged
+            // But of course it gets the data at a different rate, so we tweak the averaging ratio and frequency
+            // of updating (the latter for efficiency too).
+            var audioLevel = 0.0;
+            var avgAudioLevel = 0.0;
+
+            if (user) {
+                // we will do exponential moving average by taking some the last loudness and averaging
+                user.accumulatedLevel = AVERAGING_RATIO * (user.accumulatedLevel || 0) + (1 - AVERAGING_RATIO) * (user.audioLoudness);
+
+                // add 1 to insure we don't go log() and hit -infinity.  Math.log is
+                // natural log, so to get log base 2, just divide by ln(2).
+                audioLevel = scaleAudio(Math.log(user.accumulatedLevel + 1) / LOG2);
+
+                // decay avgAudioLevel
+                avgAudioLevel = Math.max((1 - AUDIO_PEAK_DECAY) * (user.avgAudioLevel || 0), audioLevel).toFixed(3);
+
+            }
+
+            userStore[uuid].audioLevel = audioLevel;
+            userStore[uuid].avgAudioLevel = avgAudioLevel;
+        }
+    };
+
+    var lists = {
+        // returns an array of avatarPaldata
+        allAvatars: function () {
+            return AvatarManager.getPalData().data;
+        },
+        
+        // searches the top 10 list to get the index
+        getIndexOfSettingsUser: function (uuid) {
+            if (settings.users.length) {
+                var index = settings.users.map(function (item) {
+                    return item.uuid;
+                }).indexOf(uuid);
+                return index;
+            }
+            return -1;
+        },
+
+        sortData: function () {
+            // sort by audioLevel
+            function sortNumber(a, b) {
+                return b.avgAudioLevel - a.avgAudioLevel;
+            }
+            var avatarList = Object.keys(userStore);// : lists.getAvatarsInRadius(SEARCH_RADIUS);
+            settings.users = avatarList.map(function (uuid) { return userStore[uuid]; });
```

Comment:
no space after function, put the return on a separate line(check to see if your eslint is picking this up)